Compile fails with MySQL 8.3.0 due to MySQL removing a deprecated function DBD::mysql uses

### DBD::mysql version

DBD-mysql-5.003

### MySQL client version

8.3.0

### Server version

_No response_

### Operating system version

MacOS Sonoma 14.2.1

### What happened?

Build fails with this error:

`dbdimp.c:2796:14: error: call to undeclared function 'mysql_list_fields'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]`
   ` *result= mysql_list_fields(svsock, table, NULL);`
            `.         ^`
It looks like mysql_list_fields [was removed in MySQL 8.3.0](https://github.com/mysql/mysql-server/commit/15d9b4d0e6f0a098165fb595538a9182db604778), which was released 3 weeks ago.

### Other information

_No response_

--- a/dbdimp.c
+++ b/dbdimp.c
@@ -2767,6 +2767,9 @@ my_ulonglong mysql_st_internal_execute(
   if (slen >= 11 && (!strncmp(sbuf, "listfields ", 11) ||
                      !strncmp(sbuf, "LISTFIELDS ", 11)))
   {
+    do_warn(h, JW_ERR_LIST_FIELDS,
+		    "LISTFIELDS support is deprecated and will be removed in an upcoming release");
+
     /* remove pre-space */
     slen-= 10;
     sbuf+= 10;

--- a/dbdimp.c
+++ b/dbdimp.c
@@ -2793,7 +2793,23 @@ my_ulonglong mysql_st_internal_execute(
     }
     *sbuf++= '\0';
 
-    *result= mysql_list_fields(svsock, table, NULL);
+    char* buffer = malloc(strlen(table)+20);
+    if (buffer == NULL)
+    {
+      do_error(h, JW_ERR_MEM, "Out of memory" ,NULL);
+      return -2;
+    }
+    else
+    {
+      strcpy(buffer, "SHOW COLUMNS FROM ");
+      strcat(buffer, table);
+      if (mysql_real_query(svsock, buffer, strlen(buffer)) != 0) {
+        do_error(h, JW_ERR_QUERY, "Query for columns failed" ,NULL);
+        return -2;
+      }
+      *result = mysql_store_result(svsock);
+      free(buffer);
+    }
 
     free(table);
 

--- a/lib/DBD/mysql.pm
+++ b/lib/DBD/mysql.pm
@@ -188,6 +188,8 @@ sub admin {
     my($user) = shift || '';
     my($password) = shift || '';
 
+    warn 'admin() is deprecated and will be removed in an upcoming version';
+
     $drh->func(undef, $command,
 	       $dbname || '',
 	       $host || '',

--- a/mysql.xs
+++ b/mysql.xs
@@ -100,6 +100,7 @@ _admin_internal(drh,dbh,command,dbname=NULL,host=NULL,port=NULL,user=NULL,passwo
   int retval;
   MYSQL* sock;
   const char *shutdown = "SHUTDOWN";
+  const char *reload = "FLUSH PRIVILEGES";
 
   /*
    *  Connect to the database, if required.
@@ -123,7 +124,7 @@ _admin_internal(drh,dbh,command,dbname=NULL,host=NULL,port=NULL,user=NULL,passwo
   if (strEQ(command, "shutdown"))
     retval = mysql_real_query(sock, shutdown, strlen(shutdown));
   else if (strEQ(command, "reload"))
-    retval = mysql_reload(sock);
+    retval = mysql_real_query(sock, reload, strlen(reload));
   else if (strEQ(command, "createdb"))
   {
     char* buffer = malloc(strlen(dbname)+50);

